File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/shared/exceptions/UnexpectedExceptionToasterResponse.java
package com.lasthopesoftware.bluewater.shared.exceptions;

import android.content.Context;
import com.namehillsoftware.handoff.promises.response.ImmediateResponse;

public class UnexpectedExceptionToasterResponse implements ImmediateResponse<Throwable, Void> {

	private final Context context;

	public UnexpectedExceptionToasterResponse(Context context) {
		this.context = context;
	}

	@Override
	public Void respond(Throwable error) {
		UnexpectedExceptionToaster.announce(context, error);

		return null;
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/shared/exceptions/UnexpectedExceptionToasterResponse.kt
package com.lasthopesoftware.bluewater.shared.exceptions

import android.content.Context
import com.lasthopesoftware.bluewater.shared.lazyLogger
import com.namehillsoftware.handoff.promises.response.ImmediateResponse

private val logger by lazyLogger<UnexpectedExceptionToasterResponse>()

class UnexpectedExceptionToasterResponse(private val context: Context) : ImmediateResponse<Throwable?, Unit> {
    override fun respond(error: Throwable?) {
		logger.error("An unexpected exception occurred, announcing it.", error)
		if (error != null)
        	UnexpectedExceptionToaster.announce(context, error)
    }
}
